app/models/candidate_interface/personal_details_form.rb

```diff
@@ -8,8 +8,37 @@ class PersonalDetailsForm
                   :english_main_language,
                   :english_language_details, :other_language_details
 
+    validates :first_name, :last_name, :english_main_language,
+              :first_nationality,
+              presence: true
+
+    validates :first_name, :last_name,
+              length: { maximum: 100 }
+
+    validate :date_of_birth_cannot_be_in_the_future
+
+    validates :first_nationality, :second_nationality,
+              inclusion: { in: NATIONALITIES, allow_blank: true }
+
+    validates :english_language_details, :other_language_details,
+              word_count: { maximum: 200 }
+
     def name
       "#{first_name} #{last_name}"
     end
+
+    def date_of_birth
+      Date.new(*[year, month, day].map(&:to_i))
+    rescue ArgumentError, NoMethodError
```

Comment:
Missed this in the review - I'd rather not `rescue` these errors, is there an alternative?

Reply:
Fix'd using `date_valid?`, also added more tests!